drivers/pwm/pwm_led_esp32.c

```diff
@@ -356,11 +357,15 @@ PINCTRL_DT_INST_DEFINE(0);
 	#define CLOCK_SOURCE	LEDC_SCLK
 #endif
 
+/* Fix max timer resolution as 1 bit less (by default) to allow 100% duty */
+#define CHECK_MAX_RES(resval) (resval ? resval : (SOC_LEDC_TIMER_BIT_WIDTH - 1))
+
 #define CHANNEL_CONFIG(node_id)                                                \
 	{                                                                      \
 		.idx = DT_REG_ADDR(node_id),                                   \
 		.channel_num = DT_REG_ADDR(node_id) % 8,                       \
 		.timer_num = DT_PROP(node_id, timer),                          \
+		.max_res = CHECK_MAX_RES(DT_PROP(node_id, max_res)),           \
```

Comment:
I think you could use also DT_PROP_OR(node_id, max_res, SOC_LED_TIMER_BIT_WIDTH-1) instead of defining a new macro.